subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedCustomTaskExecutionIntegrationTest.groovy

```diff
@@ -648,6 +649,36 @@ class CachedCustomTaskExecutionIntegrationTest extends AbstractIntegrationSpec i
         skippedTasks.sort() == [":consumer", ":producer"]
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/3043")
+    def "URL-quoted characters in file names are handled properly"() {
+        def weirdOutputPath = 'build/bad&dir/bad! <Dezső> %20.txt'
+        def expectedOutput = file(weirdOutputPath)
+        buildFile << """
+            task weirdOutput {
+                outputs.dir("build")
+                outputs.cacheIf { true }
+                doLast {
+                    mkdir file('$weirdOutputPath').parentFile
+                    file('$weirdOutputPath').text = "Data"
+                }
+            }
+        """
+
+        when:
+        withBuildCache().succeeds "weirdOutput"
+        then:
+        executedAndNotSkipped ":weirdOutput"
+        expectedOutput.file
+
+        when:
+        cleanBuildDir()
+        withBuildCache().succeeds "weirdOutput"
+        then:
+        skipped ":weirdOutput"
+        expectedOutput.parentFile.listFiles().each { println it }
```

Comment:
I guess this is a leftover from debugging. Please remove it.

Reply:
Fixed